R/generate.R

```diff
@@ -0,0 +1,55 @@
+int_types <- c("integer", "smallint")
+logi_types <- c("boolean")
+num_types <- c("date", "numeric", "real")
+date_types <- c("timestamp without time zone", "timestamp with time zone")
+
+generate_type_check <- function(column_name, data_type, is_nullable) {
+  r_type <- ifelse(data_type %in% int_types, "\"integer\"",
+                   ifelse(data_type %in% logi_types, "\"logical\"",
+                          ifelse(data_type %in% num_types, "\"double\"", "\"character\"")))
+
+  type_check <- paste0("typeof(", column_name, ") == ", r_type)
+  type_check <- ifelse(data_type %in% date_types, paste0("inherits(", column_name, ", \"POSIXct\")"), type_check)
+
+  na_check <- ifelse(is_nullable == "YES", paste0("is.na(", column_name, ") || "), "")
```

Comment:
`is.na` is probably not what you want here; I think you want to test:

```
typeof(x) == expected && !any(is.na(x))
```

for non-nullable types (so the inverse of your check here).  If you wanted too allow a nullable thing to have loose typing you could do

```
typeof(x) == expected || all(is.na(x))
```

to catch the case where someone wrote `id = NA` and ended up with a logical column.  We usually end up with typed NA values (e.g., `NA_real_` or `NA_character_`) because it makes things more self-documenting and avoids a class of bugs